### DISCUSSION TITLE
Cant access top level module from sub module

### DISCUSSION THREAD

**Discussion by tonyhart7** [fastapi/fastapi#6472] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Response, status, HTTPException, Depends, APIRouter
from sqlalchemy.orm.session import Session

from app import models, schemas
from app.database import  get_db
```

### Description

I cant access top level module from submodule

this is my structure 
~~~~
├─ app
├─main.py
├─ models.py
├─ schemas.py
|───router
│    ├─── __init__.py
│    ├─── post.py
│    └─── user.py
~~~~

I want to access models and schemas for my post and user in router 
but I cant import all module from app 
I using relative and absolute import and yet python cant find all that 

I believe import using the correct way as I following tutorial on official doc and reproduce the same structure from this https://github.com/tiangolo/full-stack-fastapi-postgresql

### Operating System

Windows

### Operating System Details

Windows 11

### FastAPI Version

0.70.1

### Python Version

Python 3.10.0

**Comment by DavidSantacruzR** [comment]
Hi there, from what I've been researching seems that python3 doesn't allow to directly import as you have. Instead you might want to use the sys module or environment variables (which I believe is the best option) to optimize your imports.

Then you can use the import method as usual.

Try this and let me know if this works:

import sys

#Set the path to where the modules you're importing.
sys.path.insert(0, 'path to your App directory')

#Here fastApi imports.

from app import models, schemas
from app.database import  get_db

**Comment by dabula-s** [comment]
try to check `PYTHONPATH` to be sure it's correct

**Comment by tonyhart7** [comment]
> Hi there, from what I've been researching seems that python3 doesn't allow to directly import as you have. Instead you might want to use the sys module or environment variables (which I believe is the best option) to optimize your imports.
> 
> Then you can use the import method as usual.
> 
> Try this and let me know if this works:
> 
> import sys
> 
> #Set the path to where the modules you're importing. sys.path.insert(0, 'path to your App directory')
> 
> #Here fastApi imports.
> 
> from app import models, schemas from app.database import get_db

where I put those sys import ?? in main file or I must that on before init file ?

**Comment by tonyhart7** [comment]
> try to check `PYTHONPATH` to be sure it's correct

I already did that 

Yes is must be windows problem (windows sucks) because I dnt see it in anybody code

**Comment by DavidSantacruzR** [comment]
Could be even in the init file. However, I’d say it’s best practice to have it in the scripts where you need them.

Try if it works in any file from your router dir.

**Comment by tonyhart7** [comment]
> Could be even in the init file. However, I’d say it’s best practice to have it in the scripts where you need them.
> 
> Try if it works in any file from your router dir.

Yeah its works, iam using this 
~~~
import sys
import os
sys.path.append(os.path.dirname(os.getcwd()))
~~~

Iam tried to pip install -e . and make my project to the ediable state but I cant find the import 

apreciate it

**Comment by DavidSantacruzR** [comment]
Something that you can also try is setting the path as an environment variable and accessing it using the OS library.
